src/vos/tests/evt_ctl.c

```diff
@@ -1440,6 +1448,173 @@ test_evt_iter_delete_internal(void **state)
 	assert_int_equal(rc, 0);
 }
 
+static void
+test_evt_variable_record_size_internal(void **state)
+{
+	struct test_arg		*arg = *state;
+	daos_handle_t		 toh;
+	struct evt_entry_in	 entry = {0};
+	int			 rc, count;
+	int			 epoch;
+	const int		val[] = {D_1K_SIZE,
+					D_16K_SIZE};
+	uint64_t		data_size;
+	char                     *data;
+
+	rc = evt_create(ts_feats, 13, arg->ta_uma, arg->ta_root,
+		DAOS_HDL_INVAL, &toh);
+	assert_int_equal(rc, 0);
+	for (count = 0; count < sizeof(val)/sizeof(int); count++) {
+		/* Try to insert a bunch of entries with variable data sizes */
+		data_size = val[count];
+		D_ALLOC(data, data_size);
+		strcpy(data, "EVTree: Test Variable Record Size");
+		for (epoch = 1; epoch < 3 ; epoch++) {
+			entry.ei_rect.rc_ex.ex_lo = epoch;
+			entry.ei_rect.rc_ex.ex_hi = epoch + data_size;
+			entry.ei_rect.rc_epc = epoch;
+			memset(&entry.ei_csum, 0, sizeof(entry.ei_csum));
+			entry.ei_ver = 0;
+			entry.ei_inob = data_size;
```

Comment:
style: it would be nice if you could add some blank lines in your code, for example, at here after a few lines of assignments for the same data structure, so reader can easily remember the code position while reading.

Reply:
No problem. I can add blank line after the assignments.

Reply:
Done.